lib/api/apiUtils/object/lifecycle.js

```diff
@@ -46,6 +46,15 @@ function processCurrents(bucketName, listParams, list) {
             DataStoreName: v.dataStoreName,
             ListType: CURRENT_TYPE,
         };
+
+        // NOTE: The current versions listed to be lifecycle should include version id
```

Comment:
I think the inclusion of `VersionId` should not depend on whether the bucket is versioned or not, but only whether the API call is supposed to return them. At least this is what the AWS API does. In other words, we could return `VersionId: 'null'` for all objects if the bucket is not versioned.

Reply:
If the bucket is not versioned, AWS returns objects without VersionID.
This version id is used during lifecycle transition where the object transition processor will do a putMetadata request specifying as parameters the bucket name, key name and version id to update the object metadata `dataStoreName`.
If the bucket is not versioned and the `putMetadata` call specifies the "null" version id, a new "null" version will be created instead of updating the master version.

Reply:
> If the bucket is not versioned, AWS returns objects without VersionID.

It does :wink: 

```
jonathan@jonathan-XPS-13-9380:~/git/home/ironman/S3C-7352$ aws --profile scality s3 mb s3://jonathan-scality-nonversioned-bucket
make_bucket: jonathan-scality-nonversioned-bucket
jonathan@jonathan-XPS-13-9380:~/git/home/ironman/S3C-7352$ aws --profile scality s3 cp /etc/hosts s3://jonathan-scality-nonversioned-bucket/hosts1
upload: ../../../../../../etc/hosts to s3://jonathan-scality-nonversioned-bucket/hosts1
jonathan@jonathan-XPS-13-9380:~/git/home/ironman/S3C-7352$ aws --profile scality s3api list-object-versions --bucket jonathan-scality-nonversioned-bucket
{
    "Versions": [
        {
            "ETag": "\"e171fed98b302ddc8b64aa03a56b2af4\"",
            "Size": 376,
            "StorageClass": "STANDARD",
            "Key": "hosts1",
            "VersionId": "null",
            "IsLatest": true,
            "LastModified": "2023-04-04T21:48:21.000Z",
            "Owner": {
                "DisplayName": "ironman-mgr",
                "ID": "b9e4605d555d11651588cfec37789266a156211aeaa7173a5abb82229348c8d9"
            }
        }
    ]
}
```

> If the bucket is not versioned and the putMetadata call specifies the "null" version id, a new "null" version will be created instead of updating the master version.

I believe the correct way in our API to do this is to pass an empty version ID with `&versionId=`. So if it is the null version, write it with an empty version ID.

I think it will be slightly cleaner to follow this scheme, as it is more aligned with the standard API. Also don't forget that the null version may not be the latest version, so updating the master is not always the right thing to do when updating a null version (but it is correct when writing a new null version).

Reply:
If the bucket is not versioned, BB Lifecycle Bucket Processor does a list-objects, and no VersionId is returned.
To be compatible with the old Bucket Processos implementation, I added this check.

Also, to remove this check we need to make sure Backbeat handles "null" version appropriately, https://scality.atlassian.net/browse/CLDSRV-374 
I will first focus on the CLDSRV-374  issue.